instrumentation/opentelemetry-instrumentation-httpx/README.rst

```diff
@@ -0,0 +1,96 @@
+OpenTelemetry HTTPX Instrumentation
+===================================
+
+|pypi|
+
+.. |pypi| image:: https://badge.fury.io/py/opentelemetry-instrumentation-httpx.svg
+   :target: https://pypi.org/project/opentelemetry-instrumentation-httpx/
+
+This library allows tracing HTTP requests made by the
+`httpx <https://www.python-httpx.org/>`_ library.
+
+Installation
+------------
+
+::
+
+     pip install opentelemetry-instrumentation-httpx
+
+
+Usage
+-----
+
+Instrumenting all clients
+*************************
+
+When using the instrumentor, all newly created clients will automatically trace
+requests.
+
+.. code-block:: python
+
+     import httpx
+     from opentelemetry.instrumentation.httpx import HTTPXClientInstrumentor
+
+     url = "https://httpbin.org/get"
+     HTTPXClientInstrumentor().instrument()
+
+     with httpx.Client() as client:
+          response = client.get(url)
+
+     async with httpx.AsyncClient() as client:
+          response = await client.get(url)
+
+Uninstrument
+^^^^^^^^^^^^
+
+If you need to uninstrument clients, there are two options available.
+
+.. code-block:: python
+
+     import httpx
+     from opentelemetry.instrumentation.httpx import HTTPXClientInstrumentor
+
+     HTTPXClientInstrumentor().instrument()
+     client = httpx.Client()
+
+     # Uninstrument a specific client
+     HTTPXClientInstrumentor.uninstrument_client(client)
```

Comment:
We should also have an example that shows how to instrument only a specific client and ideally have `instrument() + uninstrument()` and `instrument_client() + uninstrument_client()` as separate examples.

Reply:
I added additional documentation for these

Reply:
Is this documentation valid? I mean if we instrument whole library with `instrument()`, can we still uninstrument a single client with `uninstrument_client()` even if it was globally instrumented with `instrument()`? 

Not objecting to it. Just verifying if that's how it'd supposed to work.

Reply:
Yes, that works. I can adjust the documentation and tests to make that use case clearer.